modules/generator/processor/servicegraphs/servicegraphs.go

```diff
@@ -188,29 +196,51 @@ func (p *processor) consume(resourceSpans []*v1.ResourceSpans) error {
 	return nil
 }
 
-func (p *processor) Shutdown(ctx context.Context) {
+func (p *processor) Shutdown(_ context.Context) {
 	close(p.closeCh)
 }
 
 // collectEdge records the metrics for the given edge.
 // Returns true if the edge is completed or expired and should be deleted.
 func (p *processor) collectEdge(e *store.Edge) {
 	if e.IsCompleted() {
-		clientServerLabelValues := registry.NewLabelValues([]string{e.ClientService, e.ServerService})
+		values := []string{e.ClientService, e.ServerService}
+		for _, dimension := range p.cfg.Dimensions {
+			values = append(values, e.Dimensions[dimension])
+		}
```

Comment:
We can preallocate this slice with the expected length, this saves some memory allocations as we append the dimensions.

Reply:
Addressed in 309793e3fe14e9817d916b0c89d46c77f40508d5